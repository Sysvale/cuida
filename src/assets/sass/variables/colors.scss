$bg-gradient: 158.71deg, #FFFFFF -36.57%, #F1F5FB 156.92%;

$n-0: #ffffff;
$n-10: #FAFCFE;
$n-20: #EFF5FB;
$n-30: #E2EAF3;
$n-40: #D3DBE4;
$n-50: #BCC7D2;
$n-100: #ABB8C4;
$n-200: #9AA8B6;
$n-300: #8B99A7;
$n-400: #7C8A98;
$n-500: #566676;
$n-600: #475766;
$n-700: #36424E;
$n-800: #29333D;
$n-900: #1E252C;

$gp-50: #f3fcf8;
$gp-100: #def7eb;
$gp-200: #ABEDD9;
$gp-300: #6DDFBC;
$gp-400: #2AC092;
$gp-500: #239F78;
$gp-600: #197155;
$gp-700: #145C45;

$ts-50: #F7FCFF;
$ts-100: #F0FAFF;
$ts-200: #C2EDFF;
$ts-300: #7BD0F4;
$ts-400: #3CACDD;
$ts-500: #1F86B2;
$ts-600: #18698B;
$ts-700: #13526D;

$bn-50: #F8FAFE;
$bn-100: #ecf1fe;
$bn-200: #B6D1F7;
$bn-300: #83AFEC;
$bn-400: #377EE1;
$bn-500: #206ED9;
$bn-600: #1B59B1;
$bn-700: #144385;

$in-50: #F2F4FF;
$in-100: #E5EAFF;
$in-200: #CED6FD;
$in-300: #AAB7F8;
$in-400: #6174D1;
$in-500: #5464B6;
$in-600: #3A4788;
$in-700: #353D64;

$vr-50: #FBF7FF;
$vr-100: #F8F0FF;
$vr-200: #E6D2F9;
$vr-300: #CEABED;
$vr-400: #A169D3;
$vr-500: #8F4DCC;
$vr-600: #7633B2;
$vr-700: #642B96;

$pp-50: #FFF7FC;
$pp-100: #FFEFF9;
$pp-200: #F9C8E2;
$pp-300: #EF8FC0;
$pp-400: #DD5A9C;
$pp-500: #CC3381;
$pp-600: #A52766;
$pp-700: #701A48;

$rc-50: #fef5f6;
$rc-100: #ffe6e9;
$rc-200: #FBD0D6;
$rc-300: #F797A3;
$rc-400: #F95367;
$rc-500: #E83047;
$rc-600: #D22D41;
$rc-700: #A42333;

$og-50: #FFF9F7;
$og-100: #FFF3F0;
$og-200: #FFD6CC;
$og-300: #FDAF9B;
$og-400: #FF8567;
$og-500: #F5613D;
$og-600: #D64B29;
$og-700: #AB3C21;

$al-50: #fefbf5;
$al-100: #fff1dc;
$al-200: #FFE2B8;
$al-300: #FDCD87;
$al-400: #FFB952;
$al-500: #EEA22F;
$al-600: #D38817;
$al-700: #A56A12;

$g: (
);

$g: map-merge(("50": $gp-50,
			"100": $gp-100,
			"200": $gp-200,
			"300": $gp-300,
			"400": $gp-400,
			"500": $gp-500,
			"600": $gp-600,
			"700": $gp-700,
		),
		$g);

$t: (
);

$t: map-merge(("50": $ts-50,
			"100": $ts-100,
			"200": $ts-200,
			"300": $ts-300,
			"400": $ts-400,
			"500": $ts-500,
			"600": $ts-600,
			"700": $ts-700,
		),
		$t);

$b: (
);

$b: map-merge(("50": $bn-50,
			"100": $bn-100,
			"200": $bn-200,
			"300": $bn-300,
			"400": $bn-400,
			"500": $bn-500,
			"600": $bn-600,
			"700": $bn-700,
		),
		$b);

$i: (
);

$i: map-merge(("50": $in-50,
			"100": $in-100,
			"200": $in-200,
			"300": $in-300,
			"400": $in-400,
			"500": $in-500,
			"600": $in-600,
			"700": $in-700,
		),
		$i);

$v: (
);

$v: map-merge(("50": $vr-50,
			"100": $vr-100,
			"200": $vr-200,
			"300": $vr-300,
			"400": $vr-400,
			"500": $vr-500,
			"600": $vr-600,
			"700": $vr-700,
		),
		$v);

$p: (
);

$p: map-merge(("50": $pp-50,
			"100": $pp-100,
			"200": $pp-200,
			"300": $pp-300,
			"400": $pp-400,
			"500": $pp-500,
			"600": $pp-600,
			"700": $pp-700,
		),
		$p);

$r: (
);

$r: map-merge(("50": $rc-50,
			"100": $rc-100,
			"200": $rc-200,
			"300": $rc-300,
			"400": $rc-400,
			"500": $rc-500,
			"600": $rc-600,
			"700": $rc-700,
		),
		$r);

$o: (
);

$o: map-merge(("50": $og-50,
			"100": $og-100,
			"200": $og-200,
			"300": $og-300,
			"400": $og-400,
			"500": $og-500,
			"600": $og-600,
			"700": $og-700,
		),
		$o);

$a: (
);

$a: map-merge(("50": $al-50,
			"100": $al-100,
			"200": $al-200,
			"300": $al-300,
			"400": $al-400,
			"500": $al-500,
			"600": $al-600,
			"700": $al-700,
		),
		$a);

$n: (
);

$n: map-merge(("0": $n-0,
			"10": $n-10,
			"20": $n-20,
			"30": $n-30,
			"40": $n-40,
			"50": $n-50,
			"100": $n-100,
			"200": $n-200,
			"300": $n-300,
			"400": $n-400,
			"500": $n-500,
			"600": $n-600,
			"700": $n-700,
			"800": $n-800,
			"900": $n-900,
		),
		$n);

$palete: (
);

$palete: map-merge(('PiccoloGreen': $g,
			'SulivanTurquoise': $t,
			'NocturneBlue': $b,
			'NightWingIndigo': $i,
			'RavenViolet': $v,
			'PeppaPink': $p,
			'CarmenRed': $r,
			'GokuOrange': $o,
			'LisaAmber': $a,
			'Neutrals': $n,
			'bg-gradient': $bg-gradient,
		),
		$palete);
