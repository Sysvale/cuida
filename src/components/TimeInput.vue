<template>
	<div>
		<label
			:for="id"
			class="input__label"
		>
			{{ label }}

			<span
				v-if="required"
				class="input--required"
			>
				*
			</span>
		</label>
		<div
			:id="id"
			class="input__container"
		>
			<div>
				<input
					:id="`${id}-start-hour`"
					v-model="startHour"
					v-facade="'##'"
					:class="computedInputClass"
					:disabled="disabled"
					type="text"
					min="0"
					max="23"
					step="1"
					placeholder="00"
					@keyup.up="startHour++"
					@keyup.down="startHour > 0 ? startHour-- : null"
					@input="handleTimeInput"
				>
				:
				<input
					:id="`${id}-start-minute`"
					v-model="startMinute"
					v-facade="'##'"
					:class="computedInputClass"
					:disabled="disabled"
					type="text"
					min="0"
					max="59"
					step="1"
					placeholder="00"
					@keyup.up="startMinute++"
					@keyup.down="startMinute > 0 ? startMinute-- : null"
					@input="handleTimeInput"
				>
			</div>
			<div
				v-if="mode === 'range'"
				class="input__separator"
			>
				Até
			</div>
			<div v-if="mode === 'range'">
				<input
					:id="`${id}-end-hour`"
					v-model="endHour"
					v-facade="'##'"
					:class="computedInputClass"
					:disabled="disabled"
					type="text"
					min="0"
					max="23"
					step="1"
					placeholder="00"
					@keyup.up="endHour++"
					@keyup.down="endHour > 0 ? endHour-- : null"
					@input="handleTimeInput"
				>
				:
				<input
					:id="`${id}-end-minute`"
					v-model="endMinute"
					v-facade="'##'"
					:class="computedInputClass"
					:disabled="disabled"
					type="text"
					min="0"
					max="59"
					step="1"
					placeholder="00"
					@keyup.up="endMinute++"
					@keyup.down="endMinute > 0 ? endMinute-- : null"
					@input="handleTimeInput"
				>
			</div>
		</div>
		<div
			v-if="state === 'invalid'"
			class="input__message"
		>
			{{ errorMessage }}
		</div>
	</div>
</template>

<script>
import { facade } from 'vue-input-facade';
import { Interval, DateTime } from 'luxon';
import { hourFormat, isAfter } from '../utils/validators/time';

const valueValidator = (value) => {
	if (value === '' || value === []) {
		return true;
	}

	if (typeof value === 'string') {
		return hourFormat(value);
	}

	return value.length === 2
		&& isAfter(value[1], value[0]);
};

export default {
	directives: {
		facade,
	},

	props: {
		/**
		* Prop utilizada como v-model. Define o horário exibido.
		*
		* Modo `single`: Deve ser enviada como uma String contendo o horário.
		*
		* Modo `range`: Deve ser enviada como um Array contendo dois horários (inicial e final).
		*
		* Pode ser enviada como um Array vazio ou String vazia.
		*
		* Todos os horários devem estar no formato `HH:mm`.
		*
		*/
		modelValue: {
			type: [String, Array],
			default: '',
			validator: valueValidator,
		},
		/**
		 * O id a ser utilizado pelo elemento HTML.
		 */
		id: {
			type: String,
			default: 'time-input',
		},
		/**
		 * A label a ser exibida acima do input.
		 */
		label: {
			type: String,
			required: true,
		},
		/**
		 * Propriedade utilizada para definir o estilo da label do componente.
		 * Em caso positivo, exibe um * indicando que o campo é de preenchimento obrigatório.
		 */
		required: {
			type: Boolean,
			default: false,
		},
		/**
		 * Propriedade utilizada para definir o desabilitado do componente
		 */
		disabled: {
			type: Boolean,
			default: false,
		},
		/**
		 * Propriedade utilizada para definir o modo de exibição do componente.
		 *
		 * `single`: Apenas um input de tempo;
		 *
		 * `range`: Dois inputs de tempo (início e fim).
		 */
		mode: {
			type: String,
			default: 'single',
			validator: (value) => ['single', 'range'].includes(value),
		},
		/**
		 * Propriedade utilizada para indicar o estado de validação do componente.
		 */
		state: {
			type: String,
			default: 'default',
			validator: (value) => ['default', 'valid', 'invalid'].includes(value),
		},
		/**
		 * Mensagem a ser exibida em caso de estado inválido.
		 */
		errorMessage: {
			type: String,
			default: 'Horário inválido',
		},
	},

	data() {
		return {
			startHour: '',
			startMinute: '',
			endHour: '',
			endMinute: '',
		};
	},

	computed: {
		computedInputClass() {
			switch (this.state) {
				case 'valid':
					return 'input__time--valid';
				case 'invalid':
					return 'input__time--invalid';
				default:
					return 'input__time';
			}
		},
	},

	watch: {
		modelValue(newValue, oldValue) {
			if (newValue === oldValue || newValue == null) {
				return;
			}

			this.buildTimeElements(newValue);
		},
	},

	mounted() {
		this.buildTimeElements(this.modelValue);
	},

	methods: {
		handleTimeInput() {
			this.startHour = this.startHour ? this.resolveTimeElement(this.startHour, 23) : '';
			this.startMinute = this.startMinute ? this.resolveTimeElement(this.startMinute, 59) : '';
			this.endHour = this.endHour ? this.resolveTimeElement(this.endHour, 23) : '';
			this.endMinute = this.endMinute ? this.resolveTimeElement(this.endMinute, 59) : '';

			if (
				!(this.startHour && this.startMinute)
				|| (this.range && !(this.endHour && this.endMinute))
			) {
				this.$emit('update:modelValue', null);
				return;
			}

			if (this.mode === 'range') {
				this.resolveRangeInterval();
				return;
			}

			/**
			* Evento indicando que o input foi preenchido.
			* Retorna uma string com o horário, caso o componente esteja em modo `single`,
			* ou um array contendo horários inicial e final, quando em modo `range`.
			*
			* Em caso de o valor do campo estar inválido, o evento é emitido com valor `null`,
			* leve isto em consideração em possíveis formatações.
			*
			* As datas são retornadas sempre no formato `HH:mm`.
			* @event update:modelValue
			* @type {Event}
			*/
			this.$emit('update:modelValue', `${this.startHour}:${this.startMinute}`);
		},

		resolveRangeInterval() {
			const interval = Interval.fromDateTimes(
				DateTime.fromFormat(`${this.startHour}:${this.startMinute}`, 'HH:mm'),
				DateTime.fromFormat(`${this.endHour}:${this.endMinute}`, 'HH:mm'),
			);

			if (interval.invalid) {
				this.$emit('update:modelValue', null);
				return;
			}

			this.$emit('update:modelValue', [
				interval.start.toFormat('HH:mm'),
				interval.end.toFormat('HH:mm'),
			]);
		},

		formatTimeElement(element) {
			return parseInt(element, 10).toLocaleString(undefined, {
				minimumIntegerDigits: 2,
				maximumFractionDigits: 0,
			});
		},

		resolveTimeElement(element, maxValue) {
			const internalElement = this.formatTimeElement(element);

			if (internalElement > maxValue) {
				return maxValue;
			}

			if (internalElement < 0) {
				return '00';
			}

			return internalElement;
		},

		buildTimeElements(time) {
			if (this.mode === 'single') {
				if (typeof time !== 'string') {
					return;
				}

				[this.startHour, this.startMinute] = time.split(':');
				return;
			}

			if (typeof time === 'string') {
				this.startHour = '';
				this.startMinute = '';
				this.endHour = '';
				this.endMinute = '';

				return;
			}

			[this.startHour, this.startMinute] = time[0].split(':');
			[this.endHour, this.endMinute] = time[1].split(':');
		},
	},
};
</script>

<style lang="scss" scoped>
@import '../assets/sass/tokens.scss';

.input {
	&__container {
		color: $n-400;
		display: flex;
		align-items: center;
	}

	&__separator {
		@include caption;
		margin: mx(3);
		color: $n-600;
	}

	&__time {
		@include caption;
		border: 1px solid $n-50;
		border-radius: $border-radius-lil;
		width: 36px;
		height: 40px;
		text-align: center;
		color: $n-600;

		&::placeholder {
			color: $n-200;
		}

		&:focus-visible {
			outline-color: $bn-300;
		}

		&--valid {
			@extend .input__time;
			border: 1px solid $gp-500;

			&:focus-visible {
				outline-color: $gp-500;
			}
		}

		&--invalid {
			@extend .input__time;
			border: 1px solid $rc-500;

			&:focus-visible {
				outline-color: $rc-500;
			}
		}
	}

	&--required {
		color: $rc-600;
		font-weight: $font-weight-semibold;
	}

	&__label {
		@include label;
	}

	&__message {
		@include caption;
		color: $rc-500;
		margin: mt(2);
	}
}

/* Chrome, Safari, Edge, Opera */
input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

/* Firefox */
input[type=number] {
  -moz-appearance: textfield;
}
</style>
