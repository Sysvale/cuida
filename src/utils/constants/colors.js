export const colors = [
	{
		name: 'turquoise',
		hex: '#3CACDD', // $ts-400
		lighterHex: '#C2EDFF', // $ts-200
	},
	{
		name: 'teal',
		hex: '#3CACDD', // $ts-400
		lighterHex: '#C2EDFF', // $ts-200
	},
	{
		name: 'green',
		hex: '#2AC092', // $gp-400
		lighterHex: '#ABEDD9', // $gp-200
	},
	{
		name: 'blue',
		hex: '#377EE1', // $bn-400
		lighterHex: '#B6D1F7', // $bn-200
	},
	{
		name: 'indigo',
		hex: '#6174D1', // $in-400
		lighterHex: '#CED6FD', // $in-200
	},
	{
		name: 'violet',
		hex: '#A169D3', // $vr-400
		lighterHex: '#E6D2F9', // $vr-200
	},
	{
		name: 'pink',
		hex: '#DD5A9C', // $pp-400
		lighterHex: '#F9C8E2', // $pp-200
	},
	{
		name: 'red',
		hex: '#F95367', // $rc-400
		lighterHex: '#FBD0D6', // $rc-200
	},
	{
		name: 'orange',
		hex: '#FF8567', // $og-400
		lighterHex: '#FFD6CC', // $og-200
	},
	{
		name: 'amber',
		hex: '#FFB952', // $al-400
		lighterHex: '#FFE2B8', // $al-200
	},
	{
		name: 'gray',
		hex: '#9AA8B6', // $n-200
		lighterHex: '#ABB8C4', // $n-100
	},
];

export const colorOptions = colors.map((color) => color.name);

export const colorHexCode = (value) => {
	const foundColor = colors.filter((color) => color.name === value)[0];
	return foundColor ? foundColor.hex : '';
};

export const colorLighterHexCode = (value) => {
	const foundColor = colors.filter((color) => color.name === value)[0];
	return foundColor ? foundColor.lighterHex : '';
};
