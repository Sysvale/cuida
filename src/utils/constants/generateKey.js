export default function generateKey(length = 8) {
	let result = '';
	const characters =
		'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
	const charactersLength = characters.length;
	for (let i = 0; i < length; i += 1) {
		result += characters.charAt(Math.floor(Math.random() * charactersLength));
	}
	return result;
}
