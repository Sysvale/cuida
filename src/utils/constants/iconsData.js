export const iconsData = [
	{
		categoryName: 'Actions',
		icons: [
			'plus-outline',
			'minus-outline',
			'x-outline',
			'plus-circle-outline',
			'minus-circle-outline',
			'x-circle-outline',
			'block-outline',
			'power-outline',
			'printer-outline',
			'bookmark-outline',
			'trash-outline',
			'search-outline',
			'upload-outline',
			'download-outline',
			'edit-outline',
			'copy-outline',
			'tag-outline',
			'bullseye-outline',
			'scope-outline',
			'lifebuoy-outline',
			'trophy-outline',
			'loop-outline',
		],
	},
	{
		categoryName: 'Files & documents',
		icons: [
			'document-outline',
			'document-text-outline',
			'document-texts-outline',
			'clipboard-outline',
			'clipboard-text-outline',
			'folder-outline',
			'folder-open-outline',
			'briefcase-outline',
			'layers-outline',
			'subtask-outline',
			'box-outline',
			'package-outline',
			'attachment-clip-outline',
			'link-outline',
			'upload-cloud-outline',
			'download-cloud-outline',
		],
	},
	{
		categoryName: 'Type & content',
		icons: [
			'heading-1-outline',
			'heading-2-outline',
			'heading-3-outline',
			'text-outline',
			'divider-outline',
			'number-outline',
			'long-text-outline',
			'short-text-outline',
			'at-outline',
			'hashtag-outline',
			'bullet-list-outline',
		],
	},
	{
		categoryName: 'Notifications',
		icons: [
			'notification-bell-outline',
			'notification-bell-timer-outline',
			'alert-outline',
			'check-circle-outline',
			'check-outline',
			'info-outline',
			'help-outline',
			'warning-outline',
			'mail-outline',
			'checkbox-checked-outlined',
			'single-select-outline',
			'flag-outline',
			'sparks-outline',
			'lamp-outline',
			'lamp-on-outline',
		],
	},
	{
		categoryName: 'Menu & navigation',
		icons: [
			'home-outline',
			'dashboard-outline',
			'grid-outline',
			'list-outline',
			'menu-outline',
			'filter-outline',
			'more-horizontal-outline',
			'more-vertical-outline',
			'reorder-outline',
			'sidebar-expand-outline',
			'sidebar-collapse-outline',
		],
	},
	{
		categoryName: 'Arrows',
		icons: [
			'arrow-up-circle-outline',
			'arrow-down-circle-outline',
			'arrow-right-circle-outline',
			'arrow-left-circle-outline',
			'caret-up-outline',
			'caret-down-outline',
			'caret-right-outline',
			'caret-left-outline',
			'open-in-new-tab-outline',
			'logout-outline',
			'login-outline',
			'rotate-left-outline',
			'rotate-right-outline',
			'loading-right-outline',
			'loading-left-outline',
			'refresh-outline',
			'arrow-left-outline',
			'arrow-up-outline',
			'arrow-right-outline',
			'arrow-down-outline',
			'swap-horizontal-arrows-outline',
			'swap-vertical-arrows-outline',
			'sort-ascending-duotone',
			'sort-descending-duotone',
			'expand-outline',
			'minimize-outline',
			'move-outline',
			'undo-outline',
			'rendo-outline',
			'shuffle-outline',
			'unfold-vertical-outline',
			'unfold-horizontal-outline',
			'shrink-outline',
			'maximize-outline',
			'dropdown-outline',
		],
	},
	{
		categoryName: 'Multimedia',
		icons: [
			'translate-outline',
			'play-outline',
			'pause-outline',
			'play-circle-outline',
			'pause-circle-outline',
			'stop-circle-outline',
			'image-outline',
			'videocam-outline',
			'bolt-outline',
			'open-book-outline',
			'mouse-outline',
			'monitor-outline',
			'tv-outline',
			'totem-outline',
			'volume-1-outline',
			'volume-2-outline',
			'phone-outline',
			'share-outline',
			'headphone-outline',
			'headset-outline',
		],
	},
	{
		categoryName: 'Users',
		icons: [
			'user-outline',
			'user-remove-outline',
			'user-add-outline',
			'users-outline',
		],
	},
	{
		categoryName: 'Charts',
		icons: [
			'trending-up-outline',
			'trending-down-outline',
			'pie-chart-outline',
			'chart-column-outline',
		],
	},
	{
		categoryName: 'Settings',
		icons: ['settings-outline', 'funnel-outline', 'sliders-outline'],
	},
	{
		categoryName: 'Location',
		icons: [
			'map-pin-outline',
			'map-outline',
			'pin-outline',
			'bed-outiline',
			'building-outline',
			'ticket-outline',
			'moon-outline',
			'sun-outline',
		],
	},
	{
		categoryName: 'Date & time',
		icons: [
			'calendar-clear-outline',
			'calendar-outline',
			'clock-outline',
			'history-outline',
		],
	},
	{
		categoryName: 'Security',
		icons: [
			'visibility-on-outline',
			'visibility-off-outline',
			'lock-outline',
			'unlock-outline',
			'key-outline',
			'shield-outline',
			'shield-tick-outline',
			'shield-alert-outline',
			'radar-outline',
		],
	},
	{
		categoryName: 'Social',
		icons: [
			'chatbubble-outline', 
			'chatbubbles-outline',
			'trophy-outline',
			'weight-outline',
			'fire-outline',
			'star-outline',
		],
	},
	{
		categoryName: 'Brands',
		icons: [
			'github-logo-outline',
			'gitlab-logo-outline',
			'figma-logo-outline',
			'framer-logo-outline',
		],
	},
	{
		categoryName: 'Health',
		icons: [
			'medicine-outline',
			'test-tube-outline',
			'syringe-outline',
			'tube-outline',
			'sample-container-outline',
			'stethoscope-outline',
			'heart-rate-outline',
			'ambulance-outline',
		],
	},
];
