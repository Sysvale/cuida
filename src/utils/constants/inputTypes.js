export default ['text', 'textarea', 'email', 'password', 'tel', 'number', 'search', 'url'];
