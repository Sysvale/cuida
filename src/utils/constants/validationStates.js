export default [
	'valid',
	'invalid',
	'default',
];
