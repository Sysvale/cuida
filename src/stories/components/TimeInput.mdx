import { Meta, Story, Props, Controls, Canvas, Preview } from '@storybook/blocks';
import TimeInput from '../../components/TimeInput.vue';
import * as TimeInputStories from './TimeInput.stories';

<Meta of={TimeInputStories} />

export const Template = (args) => ({
	components: { CdsTimeInput: TimeInput },
	setup() {
		return { args };
	},
	data() {
		return {
			value: '',
		};
	},
	template: /*html*/ `
		<cds-time-input
			v-bind="args"
			v-model="value"
		/>`,
	updated() {
		console.info('%cv-model: ', 'color: #2C70CD;', this.value);
	},
});

# TimeInput

### TimeInputs são componentes utilizados para inserir horários específicos no formato simples: hora e minutos.

***

<br />

## Quando usar:

* For necessário definir horários de eventos, agendamentos, alarmes e etc…

<br />

## Quando não usar:

* For necessário mostrar opções disponíveis de horários para seleção.

<br />

## Preview

<Canvas>
  <Story of={TimeInputStories.TimeInput} />
</Canvas>

<Controls />
