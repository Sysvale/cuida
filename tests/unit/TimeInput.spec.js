import { createLocalVue, shallowMount } from '@vue/test-utils';
import flushPromises from 'flush-promises';
import TimeInput from '../../src/components/TimeInput.vue';
import { colorOptions } from '../../src/utils/constants/colors';

const localVue = createLocalVue();

describe('TimeInput', () => {
	test('if component renders correctly', async () => {
		const wrapper = shallowMount(TimeInput, {
			localVue,
			propsData: {
				label: 'Test label',
			},
		});

		await flushPromises();

		expect(wrapper).toMatchSnapshot();
	});

	test('if "value" prop is validated correctly', async () => {
		expect.assertions(2);

		const valueValidator = TimeInput.props.value.validator;

		expect(valueValidator('15:15')).toBeTruthy();
		expect(valueValidator('wrong')).toBeFalsy();
	});

	test('if input event is emitted correctly', async () => {
		const wrapper = shallowMount(TimeInput, {
			localVue,
			propsData: {
				label: 'Test label',
			},
		});

		await flushPromises();

		const inputArray = wrapper.findAll('input').wrappers;

		await inputArray[0].setValue(15);

		expect(wrapper.emitted().input).toBeFalsy();

		await inputArray[1].setValue(35);

		expect(wrapper.emitted().input[0]).toContain('15:35');
	});

	test('if wrong input value is converted to correct time range on event emission', async () => {
		const wrapper = shallowMount(TimeInput, {
			localVue,
			propsData: {
				label: 'Test label',
			},
		});

		await flushPromises();

		const inputArray = wrapper.findAll('input').wrappers;

		await inputArray[0].setValue(25.3);

		expect(wrapper.emitted().input).toBeFalsy();

		await inputArray[1].setValue(-12);

		expect(wrapper.emitted().input[0]).toContain('23:00');
	});
});
