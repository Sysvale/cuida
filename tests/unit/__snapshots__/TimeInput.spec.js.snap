// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeInput if component renders correctly 1`] = `
<div><label for="time-input" class="input__label">
    Test label

    <!---->
  </label>
  <div id="time-input" class="input__container">
    <div><input type="number" min="0" max="23" step="1" placeholder="00" class="input__time">
      :
      <input type="number" min="0" max="59" step="1" pattern="[0-9]*" placeholder="00" class="input__time">
    </div>
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;
